### PR TITLE
Add styling for active submenus on rail nav

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

https://github.com/tektoncd/dashboard/issues/803

When the main nav is collapsed to its rail presentation,
ensure we mark any link / menu with an active nav item
as active. This is to help orient users and help guide
them to the relevant section in the expanded nav.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/containers/SideNav/SideNav.js [hunk 1]
```diff
@@ -97,70 +97,132 @@ class SideNav extends Component {
...
+                !!matchPath(location.pathname, {
+                  path: this.getPath(urls.conditions.all())
+                })
+              }
               to={this.getPath(urls.conditions.all())}
             >
               Conditions
             </SideNavMenuItem>
             {this.props.isTriggersInstalled && (
-              <>
```
Comment (AlanGreene) [inline src/containers/SideNav/SideNav.js:138]: Due to the way the active item detection works in the Carbon `SideNavMenu`, the `SideNavMenuItem`s must be direct children of the `SideNavMenu` component and cannot be nested in a fragment or other component. This also means we cannot use a custom component that renders the `SideNavMenuItem` to reduce the duplication.

File: src/containers/SideNav/SideNav.js [hunk 2]
```diff
@@ -97,70 +97,132 @@ class SideNav extends Component {
...
+                }
+                to={this.getPath(urls.eventListeners.all())}
+              >
+                EventListeners
+              </SideNavMenuItem>
+            )}
+            {this.props.isTriggersInstalled && (
+              <SideNavMenuItem
+                element={NavLink}
+                isActive={
```
Comment (steveodonovan) [inline src/containers/SideNav/SideNav.js:191]: Alot of duplication in these `isActive` parameters, its not big but if you want to move them to a function that takes the url. Ill leave it up to you, how it is is fine too

Comment (AlanGreene) [reply src/containers/SideNav/SideNav.js:191]: Yeah I thought about that when I discovered the [nesting issue](https://github.com/tektoncd/dashboard/pull/1855#discussion_r535180532) but figured it didn't really bring much benefit so left it. I may have a solution / workaround to the nesting issue for a follow up PR, if that doesn't pan out I'll fix this up then if that's alright.

**Review by steveodonovan — APPROVED** [review]
/approve
/lgtm

**Pull request merged**